### PR TITLE
XNU Agent reload.sh and readme improvements

### PR DESCRIPTION
**Pull request by stek29**

### CODE REVIEW

File: lib/xnu/README.md [hunk 1]
```diff
@@ -8,9 +8,9 @@ Experimental kernel driver for boosting Frida's capabilities on macOS:
 
 ## Installation
 
-1. Open `FridaKernelAgent.xcodeproj` and build it with Xcode
-2. Review `reload.sh`, adjusting it to match your system
-3. Run `reload.sh` to install and load the driver
+Run `reload.sh` to install and load the driver
+If it fails to load, set `CODE_SIGN_IDENTITY` enviroment variable to
+identity valid for Kext signing, or disable SIP
```
Comment (oleavr) [inline lib/xnu/README.md]: Sweet! How about reflowing to 72 cols, with full stops between sentences? (Which in hindsight I should have done in my list.)

File: lib/xnu/reload.sh [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
-sudo chown -R root /Library/Extensions/FridaKernelAgent.kext
-sudo chgrp -R wheel /Library/Extensions/FridaKernelAgent.kext
-sudo kextload /Library/Extensions/FridaKernelAgent.kext
+
+KEXT_NAME=FridaKernelAgent.kext
+
+sudo kextunload "/Library/Extensions/$KEXT_NAME" &>/dev/null
+sudo rm -rf "/Library/Extensions/$KEXT_NAME"
+
+BUILD_DIR="$(mktemp -d)"
```
Comment (oleavr) [inline lib/xnu/reload.sh]: Guess we should clean this up afterwards?

File: lib/xnu/reload.sh [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
+
+cd "$(dirname "$(realpath "$0")")"
+xcodebuild build\
+  CONFIGURATION_BUILD_DIR="$BUILD_DIR"\
+  CODE_SIGN_IDENTITY="$CODE_SIGN_IDENTITY"\
+  &>/dev/null
```
Comment (oleavr) [inline lib/xnu/reload.sh]: For consistency it would be good to put a space before each continuation-backslash.

File: lib/xnu/reload.sh (hunk 2 above)
Comment (stek29) [reply lib/xnu/reload.sh]: but it's a temp dir...

File: lib/xnu/reload.sh (hunk 3 above)
Comment (stek29) [reply lib/xnu/reload.sh]: I forgot sh allows it :|

File: lib/xnu/reload.sh (hunk 2 above)
Comment (oleavr) [reply lib/xnu/reload.sh]: I know, just don't like leaving stuff there. (OCD :) )

Comment (oleavr) [reply lib/xnu/reload.sh]:
```

Comment (oleavr) [reply lib/xnu/reload.sh]: Perhaps adding this right after would do the trick:
```
function cleanup {
  rm -rf "$BUILD_DIR"
}
trap cleanup EXIT
```

**Pull request merged**